ssl/statem/extensions_clnt.c

```diff
@@ -1603,10 +1621,20 @@ int tls_parse_stoc_psk(SSL *s, PACKET *pkt, unsigned int context, X509 *x,
         return 0;
     }
 
+    /*
+     * If we used the external PSK for sending early_data then s->early_secret
+     * is already set up, so don't overwrite it. Otherwise we copy the
+     * early_secret across that we generated earlier.
```

Comment:
Now I'm curious why the one generated "here" would be different from the one used for sending early data...